File: cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

// Prompt user for 'Change Owed'
{
    float n;
    do
    {
        n = get_float("Change owed: ");
    }

    // Set input conditions
    while (n < 0);

    // Represent change in cents
    n = (n * 100);

    // Round float to last interger place
    float p = roundf(n);

    // Returns minimum number of quarters
    int q = (p / 25);

    // Returns minimum number of dimes
    int r = (int) p;
    int s = (r % 25);
    int t = (s / 10);

    // Returns minimum number of nickles
    int u = (s % 10);
    int v = (u / 5);

    // Returns minimum number of pennies
    int w = (u % 5);

    // Calculate total coins required
    int x = (q + t + v + w);

    {
        printf("%d\n", x);
    }
}

File: cash/cash.py
from cs50 import get_float

# Check input
while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Convert to cents
cents = round(change, 2) * 100

# Find components
quarters = int(cents / 25)
dimes = int((cents % 25) / 10)
nickles = int(((cents % 25) % 10) / 5)
pennies = int(((cents % 25) % 10) % 5)

# Print result
print(quarters + dimes + nickles + pennies)